apps/sms/services/js/messaging/messaging_client.js

```diff
@@ -0,0 +1,50 @@
+/* global bridge
+*/
+
+(function(exports) {
+  'use strict';
+
+  /**
+   * Name of the service that is responsible for messaging service.
+   * @type {string}
+   */
+  const SERVICE_NAME = 'messaging-service';
+
+  /**
+   * Reference to active bridge client instance.
+   *
+   * @type {Client}
+   */
+  var client;
+
+  var MessagingClient = {
+    init() {
+      var serviceEndpoint = new SharedWorker(
+        '/services/js/messaging/messaging_service.js'
+      );
+
+      serviceEndpoint.port.start();
```

Comment:
**note**: I see on MDN (https://developer.mozilla.org/en-US/docs/Web/API/MessagePort/start) that MessagePort are queuing the messages. Good for us, no race is possible !